### PR TITLE
chore: pin linter dependencies and enable dependabot

### PR DESCRIPTION
**Pull request by josecorella**
*Description of changes:*
- PR pins linter dependencies
- PR enables dependabot on master branch
- any requirement files for tox were moved to the dev_requirements/ folder. This should not affect our docs.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

# Check any applicable:
- [x] Were any files moved? Moving files changes their URL, which breaks all hyperlinks to the files.

### CODE REVIEW

**Review by lavaleri — CHANGES_REQUESTED** [review]
You also need to update https://github.com/aws/aws-dynamodb-encryption-python/blob/master/.readthedocs.yaml

I would do a grep to ensure that there are not other `test/requirements.txt` or `ci-requirements.txt` or `doc/requirements.txt` referenced anywhere that needs updating

File: examples/tox.ini [hunk 1]
```diff
@@ -24,7 +24,7 @@ passenv =
     AWS_DEFAULT_REGION
 
 sitepackages = False
-deps = -rtest/requirements.txt
+deps = -rdev_requirements/test-requirements.txt
```
Comment (lavaleri) [inline examples/tox.ini:27]: I'm wondering why this wasn't caught by CI. Does our Python CI not run the examples?

Comment (josecorella) [reply examples/tox.ini:27]: yea ci does not run the examples, i'll go ahead and add them to the ci

Comment (lavaleri) [reply examples/tox.ini:27]: It looks like we test examples as part of release validation in https://github.com/aws/aws-dynamodb-encryption-python/blob/master/codebuild/release/validate.yml

However we don't run release validation in the PR CI, nor do we explicitly target pyx-examples in our CI.

Comment (josecorella) [reply examples/tox.ini:27]: latest commit tests the examples and runs them in CI, also updates the release validation codebuild step so it uses the correct requirements file.

File: codebuild/release/validate.yml [hunk 2]
```diff
@@ -21,7 +21,7 @@ phases:
   pre_build:
     commands:
       - cd examples
-      - sed -i "s/dynamodb-encryption-sdk/dynamodb-encryption-sdk==$VERSION/" test/requirements.txt
+      - sed -i "s/dynamodb-encryption-sdk/dynamodb-encryption-sdk==$VERSION/" ../../dev_requirements/test-requirements.txt
```
Comment (farleyb-amazon) [inline codebuild/release/validate.yml:24]: Is this one too many `..`s? Looks like we've only `cd`'d one level in.

Can we run a test release to make sure this is all good?

Comment (josecorella) [reply codebuild/release/validate.yml]: great catch! turns out i also needed to navigate up a folder in the `examples/tox.ini` file. I'll send the successful run over slack

File: codebuild/release/validate.yml [hunk 3]
```diff
@@ -21,7 +21,7 @@ phases:
   pre_build:
     commands:
       - cd examples
-      - sed -i "s/dynamodb-encryption-sdk/dynamodb-encryption-sdk==$VERSION/" ../../dev_requirements/test-requirements.txt
+      - sed -i "s/dynamodb-encryption-sdk/dynamodb-encryption-sdk==$VERSION/" ../dev_requirements/examples-requirements.txt
```
Comment (farleyb-amazon) [inline codebuild/release/validate.yml:24]: This `sed` command seems like it probably breaks with the pinning of dependencies. When I try it locally I get:
```
dynamodb-encryption-sdk====3.1.0
```

**Review by farleyb-amazon — APPROVED** [review]

**Review by lavaleri — APPROVED** [review]

**Pull request merged**